src/wallet/rpcdump.cpp

```diff
@@ -647,15 +640,23 @@ UniValue importwallet(const JSONRPCRequest& request)
                         fLabel = true;
                     }
                 }
+                nTimeBegin = std::min(nTimeBegin, nTime);
                 keys.push_back(std::make_tuple(key, nTime, fLabel, strLabel));
             } else if(IsHex(vstr[0])) {
                 std::vector<unsigned char> vData(ParseHex(vstr[0]));
                 CScript script = CScript(vData.begin(), vData.end());
                 int64_t birth_time = DecodeDumpTime(vstr[1]);
+                if (birth_time > 0) nTimeBegin = std::min(nTimeBegin, birth_time);
                 scripts.push_back(std::pair<CScript, int64_t>(script, birth_time));
             }
         }
         file.close();
+        if (pwallet->chain().havePruned()) {
+            Optional<int> pruned_height = locked_chain->findPruned();
+            if (pruned_height && nTimeBegin < locked_chain->getBlockTime(*pruned_height)) {
```

Comment:
should there be some kind of grace period here, because block timestamps aren't exact?

Reply:
Yeah, just use the same grace period as is used for the wallet birthday

Reply:
I was expecting this is wrong, will fix.